server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -56,4 +64,77 @@ public Subject getSubject() {
     public TokenManager getTokenManager() {
         return identityPlugin.getTokenManager();
     }
+
+    public static IdentityService getInstance() {
+        if (instance == null) {
+            new IdentityService(Settings.EMPTY, List.of());
+        }
+        return instance;
+    }
+
+    /**
+     * Sets the application list known to the IdentityService
+     * @param applicationList A list of all principals for known applications
+     */
+    public void setApplicationList(List<Principal> applicationList) {
+        this.applicationList = List.copyOf(applicationList);
+    }
+
+    /**
+     * Returns a list of the known application principals
+     */
+    public List<Principal> getApplicationList() {
+        return List.copyOf(applicationList);
+    }
+
+    /**
+     * Returns a list of the known application principals
+     */
+    public List<String> getApplicationStrings() {
+        return applicationList.stream().map(Principal::getName).collect(Collectors.toList());
+    }
+
+
+    //TODO: Find a way to combine these
+    public Set<String> getApplicationScopes(Principal principal) {
```

Comment:
There are two sources for applications, the plugin manager and the extension manager.  I think we could create a new service where these classes (plugin/extension managers) register themselves with and the identity provider uses that class to lookup the application id / scope information.  This allows additional application sources in the future without any modification to the identity service directly.  This would also move much of the following logic out of this class.

Reply:
Do you want the new class to be Scope focused i.e. `ScopeService` or generic? I am not sure what else we will want to include in it outside of the mapping.